runtime/common/Logger.cpp

```diff
@@ -7,15 +7,34 @@
  ******************************************************************************/
 
 #include "Logger.h"
+#include "Timing.h"
 #include "spdlog/sinks/stdout_color_sinks.h"
 #include <filesystem>
+#include <set>
 #include <spdlog/cfg/env.h>
 #include <spdlog/cfg/helpers.h>
 #include <spdlog/sinks/basic_file_sink.h>
 #include <spdlog/sinks/rotating_file_sink.h>
 #include <spdlog/spdlog.h>
 
 namespace cudaq {
+
+// This must be a function rather than a global variable to avoid a startup
+// ordering issue that would otherwise occur if we simply made this a global
+// variable and then accessed it in the initializeLogger function.
+// NOTE: the only time that this list should be modified is at startup time.
+static std::set<int> &g_timingList() {
+  static std::set<int> timingList;
+  return timingList;
+}
+
+bool isTimingTagEnabled(int tag) {
+  // Note: this function is called very frequently, so it needs to be fast. It
+  // is assumed that g_timingList() contains a small number of elements
+  // (typically less than 10).
```

Comment:
Reading this comment, do you know whether `unordered_set` is faster/slower in terms of lookup time?
I got mixed results googling this :-)

Reply:
I did some quick tests, and it depends on how many timing tags you have enabled. For fewer elements (i.e. like 5 elements), it seems like `std::set` is faster, so I'll probably leave it like it is for now. We can always change it later.
(As an aside, it looks like the call to `std::chrono::high_resolution_clock::now()` is about 3-4x larger than the `set` or `unordered_set` calls right now.)